backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/action/MoveOrCopyImageGroupParameters.java

```diff
@@ -155,12 +155,20 @@ public void setForceOverride(boolean value) {
         forceOverride = value;
     }
 
-    public Guid getSourceDomainId() {
-        return sourceDomainId;
+    public Guid getSourceStorageDomainId() {
+        return sourceStorageDomainId;
+    }
+
+    public void setSourceStorageDomainId(Guid sourceStorageDomainId) {
+        this.sourceStorageDomainId = sourceStorageDomainId;
+    }
+
+    public Guid getDestStorageDomainId() {
+        return getStorageDomainId();
     }
 
-    public void setSourceDomainId(Guid value) {
-        sourceDomainId = value;
+    public void setDestStorageDomainId(Guid destStorageDomainId) {
+        setStorageDomainId(destStorageDomainId);
```

Comment:
@barpavel I read that destStorageDomainId appeared in the serialized parameters but that's both good news and bad news - 
1. it's good because it is an indication that the new code was indeed used
2. it's not good because I believe that we'll have both destStorageDomainId and storageDomainId in the parameters, and you didn't mention seeing sourceStorageDomainId.

so in order to be on the safe side and minimize the amount of data we serialize how about:
1. annotate getDestStorageDomainId with JsonIgnore so that won't be serialized
2. keep getSourceDomainId but annotate it with Deprecated and annotate getSourceStorageDomainId with JsonIgnore
that way we minimize the risk of breaking compatibility and reduce the amount of fields being persisted, yet the method with more descriptive names would be used

Reply:
@ahadas 
Old `LiveMigrateDiskParameters`:
```
{
  "@class" : "org.ovirt.engine.core.common.action.LiveMigrateDiskParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "5891af7a-c6b8-46a5-99a7-bc8aa7a386f9"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "f59cd93c-2aac-11ed-886b-001a4a16041e"
    } ],
    "externalId" : "c1ad030c-9a47-432c-ac72-16f3d392a0d6",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "admin" : true,
    "group" : false
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "Unknown",
  "commandType" : "LiveMigrateDisk",
  "multipleAction" : false,
  "entityInfo" : {
    "type" : "Disk",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "61dd4aa2-f71d-43b7-b5d3-f79b495e501e"
    } ]
  },
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "9fdce835-6125-47e7-aca3-d391e16b7575",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : null,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "Command",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "2480c056-a773-49d6-a6ed-ae21c7fc052b"
  } ],
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "0ad31bc1-552f-4b67-87cf-663cfb14254b"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "feeea07e-bce3-49c3-8c25-9b0f22611204"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b2063744-519e-4d38-9b3d-d4194f2ecd6a"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "771c0741-4be3-4435-ae87-59f81cb7edeb"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "61dd4aa2-f71d-43b7-b5d3-f79b495e501e"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "87ca3f4e-268a-42b0-bbc8-5858d9383d7d"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "061fab40-9a27-4235-a036-14109bb7601a"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "UNUSED0",
  "volumeType" : "Unassigned",
  "copyVolumeType" : "SharedVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "8abb2033-e964-45a9-9bfa-be69b8d5ebeb"
  } ],
  "destImageGroupId" : null,
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : null,
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "liveDiskMigrateStage" : "IMAGE_DATA_SYNC_EXEC_END",
  "autoGeneratedSnapshotId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "e1d34dec-6a36-4a85-8efb-31d8b8399aae"
  } ],
  "vmId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "f6eabf86-8644-4337-8085-30338d20f183"
  } ],
  "sourceStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "8abb2033-e964-45a9-9bfa-be69b8d5ebeb"
  } ],
  "targetStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "feeea07e-bce3-49c3-8c25-9b0f22611204"
  } ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "sessionId" : "",
  "shouldBeLogged" : true,
  "transactionScopeOption" : "Required"
}
```

New `LiveMigrateDiskParameters`:
```
{
  "@class" : "org.ovirt.engine.core.common.action.LiveMigrateDiskParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "5891af7a-c6b8-46a5-99a7-bc8aa7a386f9"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "f59cd93c-2aac-11ed-886b-001a4a16041e"
    } ],
    "externalId" : "c1ad030c-9a47-432c-ac72-16f3d392a0d6",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "admin" : true,
    "group" : false
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "Unknown",
  "commandType" : "LiveMigrateDisk",
  "multipleAction" : false,
  "entityInfo" : {
    "type" : "Disk",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "61dd4aa2-f71d-43b7-b5d3-f79b495e501e"
    } ]
  },
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "9fdce835-6125-47e7-aca3-d391e16b7575",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : null,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "Command",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "2480c056-a773-49d6-a6ed-ae21c7fc052b"
  } ],
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "0ad31bc1-552f-4b67-87cf-663cfb14254b"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "feeea07e-bce3-49c3-8c25-9b0f22611204"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b2063744-519e-4d38-9b3d-d4194f2ecd6a"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "771c0741-4be3-4435-ae87-59f81cb7edeb"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "61dd4aa2-f71d-43b7-b5d3-f79b495e501e"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "87ca3f4e-268a-42b0-bbc8-5858d9383d7d"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "061fab40-9a27-4235-a036-14109bb7601a"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "UNUSED0",
  "volumeType" : "Unassigned",
  "copyVolumeType" : "SharedVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "8abb2033-e964-45a9-9bfa-be69b8d5ebeb"
  } ],
  "destImageGroupId" : null,
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : null,
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "liveDiskMigrateStage" : "LIVE_MIGRATE_DISK_EXEC_COMPLETED",
  "autoGeneratedSnapshotId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "e1d34dec-6a36-4a85-8efb-31d8b8399aae"
  } ],
  "vmId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "f6eabf86-8644-4337-8085-30338d20f183"
  } ],
  "destStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "feeea07e-bce3-49c3-8c25-9b0f22611204"
  } ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "sessionId" : "",
  "transactionScopeOption" : "Required",
  "shouldBeLogged" : true
}
```

Reply:
Luckily I saved the DB content, see above :)
I didn't mention `sourceStorageDomainId`, because it was there before :)
The same with `storageDomainId`.
I tried to emphasize that `targetStorageDomainId` changed to a new `destStorageDomainId`.
Do you still see that there is an issue?

Reply:
> Luckily I saved the DB content, see above :) I didn't mention `sourceStorageDomainId`, because it was there before :) 

Ah right, you checked LSM - I meant to check a command that uses MoveOrCopyImageGroupParameters so the persisted parameters would hae sourceDomainId which won't exist after the upgrade  

> The same with `storageDomainId`. I tried to emphasize that `targetStorageDomainId` changed to a new `destStorageDomainId`.

Right, because storageDomainId still exists and set/getDestStorageDomainId refers to its methods

> Do you still see that there is an issue?

note in the output above but I still think we may have an issue with MoveOrCopyImageGroupParameters or MoveDiskParameters if we rename the methods and in the output above you see the duplication:

```
"destStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "feeea07e-bce3-49c3-8c25-9b0f22611204"
  } ],
```
and
```
"storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "feeea07e-bce3-49c3-8c25-9b0f22611204"
  } ],

plus,
```
"sourceStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "8abb2033-e964-45a9-9bfa-be69b8d5ebeb"
  } ],
```
and
```
"sourceStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "8abb2033-e964-45a9-9bfa-be69b8d5ebeb"
  } ],
```
if we rename the method for clarity, we can add those methods with JsonIgnore and deprecate the previous one to be on the safe side

Reply:
@ahadas Yes, there might be an issue with a `MoveOrCopyDiskCommand` (not LSM) command :(

Reply:
@ahadas 
Hi,
You were right.
The not LSM flow has an issue with `RemoveImageCommand` that causes the space not to be released on Source Storage Domain.
That what happens after simulating the upgrade.
Old `MoveOrCopyImageGroupParameters`:
```
{
  "@class" : "org.ovirt.engine.core.common.action.MoveOrCopyImageGroupParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "56301d68-9cb1-4e72-b935-895468cf2ea0"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "37337a00-31c3-11ed-922d-001a4a16041e"
    } ],
    "externalId" : "66bb49df-c1fc-4b3d-8759-07a3941717da",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "group" : false,
    "admin" : true
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "MoveOrCopyDisk",
  "commandType" : "MoveImageGroup",
  "multipleAction" : false,
  "entityInfo" : null,
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "0908f2ad-16ae-43d6-906d-b4a19e2d96cf",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : 100,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "None",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : null,
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "80080044-ad65-484a-bf63-b6dd9c06ad2e"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "4f414373-c797-4f57-905b-babd8631bd7a"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "996fa521-3e20-4104-ba36-6d540f3cc76e"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b92f92a9-578b-489d-b0fb-13f8330770fc"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b7839ffa-6fd3-4888-b709-0b810a000c40"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b7839ffa-6fd3-4888-b709-0b810a000c40"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "973eaaea-3d97-444b-8db3-f2e67dac60ba"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "RAW",
  "volumeType" : "Preallocated",
  "copyVolumeType" : "LeafVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b035ca0a-cba4-4e98-86a5-4598e164ce29"
  } ],
  "destImageGroupId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "973eaaea-3d97-444b-8db3-f2e67dac60ba"
  } ],
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : "iSCSI_VM1_Disk1",
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "shouldBeLogged" : true,
  "transactionScopeOption" : "Required",
  "sessionId" : ""
}
```
New `MoveOrCopyImageGroupParameters`:
```
{
  "@class" : "org.ovirt.engine.core.common.action.MoveOrCopyImageGroupParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "56301d68-9cb1-4e72-b935-895468cf2ea0"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "37337a00-31c3-11ed-922d-001a4a16041e"
    } ],
    "externalId" : "66bb49df-c1fc-4b3d-8759-07a3941717da",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "group" : false,
    "admin" : true
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "MoveOrCopyDisk",
  "commandType" : "MoveImageGroup",
  "multipleAction" : false,
  "entityInfo" : null,
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "0908f2ad-16ae-43d6-906d-b4a19e2d96cf",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : 100,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "None",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : null,
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "80080044-ad65-484a-bf63-b6dd9c06ad2e"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "4f414373-c797-4f57-905b-babd8631bd7a"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "996fa521-3e20-4104-ba36-6d540f3cc76e"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b92f92a9-578b-489d-b0fb-13f8330770fc"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b7839ffa-6fd3-4888-b709-0b810a000c40"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b7839ffa-6fd3-4888-b709-0b810a000c40"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "973eaaea-3d97-444b-8db3-f2e67dac60ba"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "RAW",
  "volumeType" : "Preallocated",
  "copyVolumeType" : "LeafVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceStorageDomainId" : null,
  "destImageGroupId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "973eaaea-3d97-444b-8db3-f2e67dac60ba"
  } ],
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : "iSCSI_VM1_Disk1",
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "destStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "4f414373-c797-4f57-905b-babd8631bd7a"
  } ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "shouldBeLogged" : true,
  "transactionScopeOption" : "Required",
  "sessionId" : ""
}
```
Note that
1. There is a new `destStorageDomainId` that is equal to the old (and still existing) `storageDomainId`.
2. `sourceDomainId` disappeared (since was removed) and the new `sourceStorageDomainId` appear as `null`!!!

Result:
```
2022-09-12 09:56:30,125+03 ERROR [org.ovirt.engine.core.bll.storage.disk.image.RemoveImageCommand] (EE-ManagedScheduledExecutorService-engineScheduledThreadPool-Thread-34) [0908f2ad-16ae-43d6-906d-b4a19e2d96cf] Command 'org.ovirt.engine.core.bll.storage.disk.image.RemoveImageCommand' failed: null
2022-09-12 09:56:30,126+03 ERROR [org.ovirt.engine.core.bll.storage.disk.image.RemoveImageCommand] (EE-ManagedScheduledExecutorService-engineScheduledThreadPool-Thread-34) [0908f2ad-16ae-43d6-906d-b4a19e2d96cf] Exception: java.lang.NullPointerException
	at deployment.engine.ear.bll.jar//org.ovirt.engine.core.bll.storage.disk.image.RemoveImageCommand.performDeleteImageVdsmOperation(RemoveImageCommand.java:323)
	at deployment.engine.ear.bll.jar//org.ovirt.engine.core.bll.storage.disk.image.RemoveImageCommand.executeCommand(RemoveImageCommand.java:122)
	at deployment.engine.ear.bll.jar//org.ovirt.engine.core.bll.CommandBase.executeWithoutTransaction(CommandBase.java:1174)
.........
2022-09-12 09:56:39,091+03 WARN  [org.ovirt.engine.core.dal.dbbroker.auditloghandling.AuditLogDirector] (EE-ManagedScheduledExecutorService-engineScheduledThreadPool-Thread-34) [0908f2ad-16ae-43d6-906d-b4a19e2d96cf] EVENT_ID: USER_MOVE_IMAGE_GROUP_FAILED_TO_DELETE_SRC_IMAGE(2,025), Possible failure while deleting iSCSI_VM1_Disk1 from the source Storage Domain iSCSI_SD2 during the move operation. The Storage Domain may be manually cleaned-up from possible leftovers (User:SYSTEM).
```

So the solution is:
1) Setter - remove or deprecate? If deprecate return the new or old data member?
2) Getter - remove or deprecate? If deprecate update the new or old data member?
3) Leave the old `sourceDomainId` data member, but deprecate it?

Reply:
what you care about is not the members but their setters/getters, right?
I'd go with leaving the variables as-is, deprecate their setters and getters and annotate the new getters with JsonIgnore

Reply:
@ahadas 
Since the original solution caused unexpected problems in upgrade scenarios, I after all decided to simplify the solution and to keep the original `sourceDomainId` data member in `MoveOrCopyImageGroupParameters` class. IMHO the whole `@Deprecated` & `@JsonIgnore` just not worth it (the improvement becomes less obvious).
So all I did is to align both source & destination Storage Domains' IDs getters and setters according to `MoveOrCopyImageGroupParameters` naming convention and moved them to the `MoveOrCopyImageGroupParameters` class.
I verified the following flows:
1. Non LSM flow.
Result: success.

Old `MoveOrCopyImageGroupParameters`:
```
{
  "@class" : "org.ovirt.engine.core.common.action.MoveOrCopyImageGroupParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "27e2f806-bf36-492d-86d3-e809c2e1a397"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "d57c0d7e-326a-11ed-b9e3-001a4a16041e"
    } ],
    "externalId" : "2a0ad699-18d5-4382-8ae5-7c74d60d60fa",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "group" : false,
    "admin" : true
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "MoveOrCopyDisk",
  "commandType" : "MoveImageGroup",
  "multipleAction" : false,
  "entityInfo" : null,
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "ad868915-660d-4608-85f0-6ad9f9aae155",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : 100,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "None",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : null,
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "d0041dd1-3c34-4247-8f75-cb3b9dbb54e0"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c3be8624-96b0-46b3-94ac-45c161b448b8"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c9bbf57e-4871-4c72-8f83-e313be7ea57d"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "882ac05b-28c7-4088-9ad9-2524e4b12b89"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "506bbdc6-e2a4-4e8f-aac6-4413fc45a51a"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "RAW",
  "volumeType" : "Preallocated",
  "copyVolumeType" : "LeafVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "a2948ef1-dfac-497a-8f22-190a3afb088a"
  } ],
  "destImageGroupId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "506bbdc6-e2a4-4e8f-aac6-4413fc45a51a"
  } ],
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : "iSCSI_VM1_Disk1",
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "shouldBeLogged" : true,
  "transactionScopeOption" : "Required",
  "sessionId" : ""
}
```
New `MoveOrCopyImageGroupParameters` (note the new `destDomainId` field that is equal to `storageDomainId`):
```
{
  "@class" : "org.ovirt.engine.core.common.action.MoveOrCopyImageGroupParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "27e2f806-bf36-492d-86d3-e809c2e1a397"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "d57c0d7e-326a-11ed-b9e3-001a4a16041e"
    } ],
    "externalId" : "2a0ad699-18d5-4382-8ae5-7c74d60d60fa",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "group" : false,
    "admin" : true
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "MoveOrCopyDisk",
  "commandType" : "MoveImageGroup",
  "multipleAction" : false,
  "entityInfo" : null,
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "ad868915-660d-4608-85f0-6ad9f9aae155",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : 100,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "None",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : null,
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "d0041dd1-3c34-4247-8f75-cb3b9dbb54e0"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c3be8624-96b0-46b3-94ac-45c161b448b8"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c9bbf57e-4871-4c72-8f83-e313be7ea57d"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "882ac05b-28c7-4088-9ad9-2524e4b12b89"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "506bbdc6-e2a4-4e8f-aac6-4413fc45a51a"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "RAW",
  "volumeType" : "Preallocated",
  "copyVolumeType" : "LeafVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "a2948ef1-dfac-497a-8f22-190a3afb088a"
  } ],
  "destImageGroupId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "506bbdc6-e2a4-4e8f-aac6-4413fc45a51a"
  } ],
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : "iSCSI_VM1_Disk1",
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "destDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c3be8624-96b0-46b3-94ac-45c161b448b8"
  } ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "shouldBeLogged" : true,
  "transactionScopeOption" : "Required",
  "sessionId" : ""
}
```

2. LSM flow.
Result: after upgrade simulation, the LSM flow continues, disk is moved to a new Storage Domain, but the operation is stuck with an endless `"Failed to acquire VM lock, will retry on the next polling cycle"` log message and the snapshot is not cleaned-up due to a known bug (which is not related to this change): https://bugzilla.redhat.com/2110186

Old `LiveMigrateDiskParameters`:
```
{
  "@class" : "org.ovirt.engine.core.common.action.LiveMigrateDiskParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "611ef845-bbe2-49dc-8b7a-44cbd9ad1fbe"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "d57c0d7e-326a-11ed-b9e3-001a4a16041e"
    } ],
    "externalId" : "2a0ad699-18d5-4382-8ae5-7c74d60d60fa",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "group" : false,
    "admin" : true
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "Unknown",
  "commandType" : "LiveMigrateDisk",
  "multipleAction" : false,
  "entityInfo" : {
    "type" : "Disk",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
    } ]
  },
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "0a98c535-933a-47fb-bbdc-16ecb611bc92",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : null,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "Command",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b25e9484-a467-4c3a-90d6-b51078b71a39"
  } ],
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "d0041dd1-3c34-4247-8f75-cb3b9dbb54e0"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c3be8624-96b0-46b3-94ac-45c161b448b8"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c9bbf57e-4871-4c72-8f83-e313be7ea57d"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "882ac05b-28c7-4088-9ad9-2524e4b12b89"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "ab7827bb-fcb1-4fa2-ab9b-6c2f557dc19c"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "506bbdc6-e2a4-4e8f-aac6-4413fc45a51a"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "UNUSED0",
  "volumeType" : "Unassigned",
  "copyVolumeType" : "SharedVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "a2948ef1-dfac-497a-8f22-190a3afb088a"
  } ],
  "destImageGroupId" : null,
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : null,
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "liveDiskMigrateStage" : "CLONE_IMAGE_STRUCTURE",
  "autoGeneratedSnapshotId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "fff23a6b-ddf9-42df-a63a-0287e5547a94"
  } ],
  "vmId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "3bf07659-9400-470a-abd8-a7e7b6d66cfb"
  } ],
  "sourceStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "a2948ef1-dfac-497a-8f22-190a3afb088a"
  } ],
  "targetStorageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c3be8624-96b0-46b3-94ac-45c161b448b8"
  } ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "shouldBeLogged" : true,
  "transactionScopeOption" : "Required",
  "sessionId" : ""
}
```
New `LiveMigrateDiskParameters` (note that the old `sourceStorageDomainId` that was equal to the `sourceDomainId` was removed, while `targetStorageDomainId` was renamed to `destDomainId`):
```
{
  "@class" : "org.ovirt.engine.core.common.action.LiveMigrateDiskParameters",
  "commandId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "611ef845-bbe2-49dc-8b7a-44cbd9ad1fbe"
  } ],
  "parametersCurrentUser" : {
    "@class" : "org.ovirt.engine.core.common.businessentities.aaa.DbUser",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "d57c0d7e-326a-11ed-b9e3-001a4a16041e"
    } ],
    "externalId" : "2a0ad699-18d5-4382-8ae5-7c74d60d60fa",
    "domain" : "internal-authz",
    "namespace" : "*",
    "loginName" : "admin",
    "firstName" : "admin",
    "lastName" : "",
    "department" : "",
    "email" : null,
    "note" : null,
    "groupNames" : [ "java.util.ArrayList", [ ] ],
    "groupIds" : [ "java.util.ArrayList", [ ] ],
    "name" : "admin",
    "group" : false,
    "admin" : true
  },
  "executionReason" : "REGULAR_FLOW",
  "compensationEnabled" : false,
  "compensationPhaseEndCommand" : false,
  "parentCommand" : "Unknown",
  "commandType" : "LiveMigrateDisk",
  "multipleAction" : false,
  "entityInfo" : {
    "type" : "Disk",
    "id" : [ "org.ovirt.engine.core.compat.Guid", {
      "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
    } ]
  },
  "taskGroupSuccess" : true,
  "vdsmTaskIds" : null,
  "vdsRunningOn" : null,
  "endProcedure" : "PARENT_MANAGED",
  "useCinderCommandCallback" : false,
  "correlationId" : "0a98c535-933a-47fb-bbdc-16ecb611bc92",
  "jobId" : null,
  "stepId" : null,
  "jobWeight" : null,
  "operationsJobWeight" : [ "java.util.Collections$EmptyMap", { } ],
  "lockProperties" : {
    "scope" : "Command",
    "timeoutMillis" : -1,
    "noWait" : true,
    "waitForever" : false,
    "timeout" : false
  },
  "lifeInMinutes" : null,
  "prevVdsStatus" : "Unassigned",
  "vdsId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "b25e9484-a467-4c3a-90d6-b51078b71a39"
  } ],
  "storagePoolId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "d0041dd1-3c34-4247-8f75-cb3b9dbb54e0"
  } ],
  "forceDelete" : false,
  "storageDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c3be8624-96b0-46b3-94ac-45c161b448b8"
  } ],
  "isInternal" : false,
  "quotaId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c9bbf57e-4871-4c72-8f83-e313be7ea57d"
  } ],
  "diskProfileId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "882ac05b-28c7-4088-9ad9-2524e4b12b89"
  } ],
  "phase" : "PROCESS_OVF_UPDATE_FOR_STORAGE_POOL",
  "imageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "84b80844-d5a7-4da0-99f2-44e52eb80d98"
  } ],
  "destinationImageId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "ab7827bb-fcb1-4fa2-ab9b-6c2f557dc19c"
  } ],
  "diskAlias" : "iSCSI_VM1_Disk1",
  "description" : null,
  "oldLastModifiedValue" : null,
  "vmSnapshotId" : null,
  "imageGroupID" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "506bbdc6-e2a4-4e8f-aac6-4413fc45a51a"
  } ],
  "importEntity" : false,
  "leaveLocked" : false,
  "initialSizeInBytes" : null,
  "backup" : null,
  "bitmap" : null,
  "wipeAfterDelete" : false,
  "operation" : "Move",
  "useCopyCollapse" : false,
  "volumeFormat" : "UNUSED0",
  "volumeType" : "Unassigned",
  "copyVolumeType" : "SharedVol",
  "addImageDomainMapping" : false,
  "forceOverride" : false,
  "sourceDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "a2948ef1-dfac-497a-8f22-190a3afb088a"
  } ],
  "destImageGroupId" : null,
  "revertDbOperationScope" : null,
  "shouldLockImageOnRevert" : false,
  "newAlias" : null,
  "destImages" : [ "java.util.ArrayList", [ ] ],
  "liveDiskMigrateStage" : "LIVE_MIGRATE_DISK_EXEC_COMPLETED",
  "autoGeneratedSnapshotId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "fff23a6b-ddf9-42df-a63a-0287e5547a94"
  } ],
  "vmId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "3bf07659-9400-470a-abd8-a7e7b6d66cfb"
  } ],
  "destDomainId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "c3be8624-96b0-46b3-94ac-45c161b448b8"
  } ],
  "containerId" : [ "org.ovirt.engine.core.compat.Guid", {
    "uuid" : "00000000-0000-0000-0000-000000000000"
  } ],
  "liveSnapshot" : false,
  "shouldBeLogged" : true,
  "transactionScopeOption" : "Required",
  "sessionId" : ""
}
```

Reply:
@ahadas the conversation can be resolved and the PR is ready to be merged, unless you see additional issues.